core/lib/pattern_exporter.js

```diff
@@ -14,18 +15,50 @@ var pattern_exporter = function () {
   function exportPatterns(patternlab) {
     //read the config export options
     var exportPartials = patternlab.config.patternExportPatternPartials;
+    var exportAll = patternlab.config.patternExportAll;
+
+    if (exportAll) {
+      for (var i = 0; i < patternlab.patterns.length; i++) {
+        if (!patternlab.patterns[i].patternPartial.startsWith('-')) {
+          exportSinglePattern(patternlab, patternlab.patterns[i]);
+        }
+      }
+
+      return;
+    }
 
     //find the chosen patterns to export
     for (var i = 0; i < exportPartials.length; i++) {
```

Comment:
I cannot think of a need to loop through a provided array if the user has chosen to export all. I suggest making these mutually exclusive. Should be simple to add this as an else block as part of the exportAll check